### PR TITLE
Add config parser

### PR DESCRIPTION
**Pull request by arosyri**
Реалізовано функцію load_config для YAML/JSON конфігів.
Додані юніт-тести:
- Завантаження YAML
- Завантаження JSON
- Обробка помилок формату

### CODE REVIEW

File: configlib/parser.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+class ConfigParserError(Exception):
+    """Custom error for parser.py"""
+    pass
+
+def load_config(filepath: str) -> dict:
+    """Loads a configuration file in YAML or JSON format"""
+    if not os.path.exists(filepath):
+        raise ConfigParserError(f"File {filepath} does not exist")
+
+    _, ext = os.path.splitext(filepath)
```
Comment (arosyri) [inline configlib/parser.py:14]: Extension checking can be moved to a separate helper function to make the code cleaner.

File: tests/test_parser.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    json_content = '{"host": "localhost", "port": 8080}'
+    with tempfile.NamedTemporaryFile(delete=False, suffix='.json') as tmp:
+        tmp.write(json_content.encode('utf-8'))
+        tmp_path = tmp.name
+
+    result = parser.load_config(tmp_path)
+    assert result == {'host': 'localhost', 'port': 8080}
+    os.remove(tmp_path)
+
+def test_unsupported_format():
```
Comment (arosyri) [inline tests/test_parser.py:26]: There's a negative case, but it could also test an empty file.

File: configlib/parser.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import json
+import yaml
+import os
+
+class ConfigParserError(Exception):
+    """Custom error for parser.py"""
+    pass
+
+def load_config(filepath: str) -> dict:
```
Comment (arosyri) [inline configlib/parser.py:9]: The feature has a clean interface. It might be worth considering supporting TOML in the future.
